tests/ui/contracts/feature-gate-warning.rs

```diff
@@ -0,0 +1,17 @@
+//@ run-pass
+//@ compile-flags: -Zcontract-checks=yes
+
+// This test specifically checks that the contracts feature flag warning is
+// emitted, so that it can be marked as `expect`ed in other tests in order to
+// reduce duplication.
```

Comment:
Problem: this test does not do what this test says it is doing. This test is checking the `incomplete_features` lint warning, and *not* feature gating, because if it was, this test should be `//@ check-fail`.